internal/clients/vault_secrets_preview.go

```diff
@@ -61,27 +61,34 @@ func OpenVaultSecretsAppSecrets(ctx context.Context, client *Client, loc *shared
 
 	var secrets *secret_service.OpenAppSecretsOK
 	var err error
-	for attempt := 0; attempt < retryCount; attempt++ {
-		secrets, err = client.VaultSecretsPreview.OpenAppSecrets(params, nil)
-		if err != nil {
-			var serviceErr *secret_service.OpenAppSecretDefault
-			ok := errors.As(err, &serviceErr)
-			if !ok {
+	var result []*secretmodels.Secrets20231128OpenSecret
+
+	for {
+		for attempt := 0; attempt < retryCount; attempt++ {
+			secrets, err = client.VaultSecretsPreview.OpenAppSecrets(params, nil)
+			if err != nil {
+				var serviceErr *secret_service.OpenAppSecretDefault
+				ok := errors.As(err, &serviceErr)
+				if !ok {
+					return nil, err
+				}
+				if shouldRetryWithSleep(ctx, serviceErr, attempt, []int{http.StatusTooManyRequests}) {
```

Comment:
It seems that if there is an issue and we get `http.StatusTooManyRequests` consistently, we get stuck in the for ever loop. Would you think that the outer for loop should also check for a break condition to avoid getting stuck in a forever loop? Like maximum number of such error, or in a period of time.

Reply:
I have updated the code and moved the logic of returning out of inner for loop. Please let me know if this does not look right. Thanks!